rero_ils/modules/stats/api/report.py

```diff
@@ -0,0 +1,445 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2023 RERO
+# Copyright (C) 2019-2023 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""To compute the statistics from configuration."""
+
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+from elasticsearch_dsl.aggs import A
+from invenio_search.api import RecordsSearch
+
+from rero_ils.modules.documents.api import DocumentsSearch
+from rero_ils.modules.holdings.api import HoldingsSearch
+from rero_ils.modules.ill_requests.api import ILLRequestsSearch
+from rero_ils.modules.items.api import ItemsSearch
+from rero_ils.modules.items.models import ItemCirculationAction
+from rero_ils.modules.libraries.api import LibrariesSearch
+from rero_ils.modules.loans.logs.api import LoanOperationLog, \
+    LoanOperationLogsSearch
+from rero_ils.modules.locations.api import LocationsSearch
+from rero_ils.modules.patron_types.api import PatronTypesSearch
+from rero_ils.modules.patrons.api import PatronsSearch
+from rero_ils.modules.utils import extracted_data_from_ref
+
+
+class StatsReport:
+    """Statistics for report."""
+
+    def __init__(self, config):
+        """Constructor.
+
+        Set variables to create report.
+        """
+        self.config = config
+        self.is_active = config.get('is_active', False)
+        self.indicator = config['category']['indicator']['type']
+        self.period = config['category']['indicator'].get('period')
+        self.distributions = config[
+            'category']['indicator'].get('distributions', [])
+        self.org_pid = extracted_data_from_ref(config['organisation'])
```

Comment:
Very low priority :: Could use "@property" as other project resources ? Of interest ?

Reply:
Not really..